docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+            }
+          }
+        ],
+        "data_deletes": [
+          {
+            "key": "key3"
+          }
+        ]
+      }
+    ]
+  },
+  "signatures": {
+    "alice": "MEUCIQC8Zd3NihB/TveLmhk/BAlvuIjSo76fnsywvyyh3JudOgIgbujXo5cMSyBa5i/I9vM3h0HWb/3Waop56hPSRB4Jw5k="
+  }
+}' | jq . 
+
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "receipt": {
+      "header": {
+        "base_header": {
+          "number": 6,
+          "previous_base_header_hash": "TJYT7zr95D3ghtU/M+j8OStU6r8Y+XRC5xfb90jshbI=",
+          "last_committed_block_hash": "r57EtiB2XO1XlVAxBwFtn7cHP7YsfUBFi69y0UY0tw0=",
+          "last_committed_block_num": 5
+        },
+        "skipchain_hashes": [
+          "r57EtiB2XO1XlVAxBwFtn7cHP7YsfUBFi69y0UY0tw0="
+        ],
+        "tx_merkel_tree_root_hash": "dSc42JVp36bfORFO8QehzQFypEtBQfUzluT6ixhjeo4=",
+        "state_merkel_tree_root_hash": "YRsFw5PcG1XzYO5o5nJa3lVfknijRqu6cCaDF1zpPA8=",
+        "validation_info": [
+          {}
+        ]
+      }
+    }
+  },
+  "signature": "MEUCIHGNm7923sl7r3o/UBPi6IA4LhmV7vjISxB2aRTIJI8uAiEA4EKMhVJIwKMv1Nzy7zDaftEkal2wFl3pQN9heKw6tCY="
+}
+```
+## History query
+First we will query for full history of changes in value of `key1` in `bd2`. To do this, we use `/provenance/data/history/{dbname}/{key}` GET query, and user `alice` will submit it.
+**Sign query**
+```sh
+bin/signer -data '{"user_id":"alice","db_name":"db2","key":"key1"}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Signature**
+```
+MEUCIHTGSy8lJFlfRxJXEGq2gTi9czP81jwQ7vF2KdRxiigRAiEAjjCn9WSQPx6H99+EGYHyDQTNAr4O+1uhvY5eYI0ZT20=
+```
+
+**Submit query**
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEUCIHTGSy8lJFlfRxJXEGq2gTi9czP81jwQ7vF2KdRxiigRAiEAjjCn9WSQPx6H99+EGYHyDQTNAr4O+1uhvY5eYI0ZT20=" \
+     -X GET http://127.0.0.1:6001/provenance/data/history/db2/key1 | jq .
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "eXl5",
+        "metadata": {
+          "version": {
+            "block_num": 4
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true,
+              "bob": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVlIHVwZGF0ZWQ=",
+        "metadata": {
+          "version": {
+            "block_num": 6
+          },
```

Comment:
you comment on that later, but better comment first time it happens

Reply:
In this case we need to change all protobuf marshaling, because leaving empty fields in json representation in some cases and omitting in another will break all signatures. I added this explanation multiple time, first time near query sign.

Reply:
Better way is to start to count transactions in block from 1.